contrib/native/client/src/include/drill/userProperties.hpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef USER_PROPERTIES_H
+#define USER_PROPERTIES_H
+
+#include <map>
+#include "drill/common.hpp"
+
+namespace Drill{
+
+class DECLSPEC_DRILL_CLIENT DrillUserProperties{
+    public:
+        static const std::map<std::string, uint32_t> USER_PROPERTIES;
+
+        DrillUserProperties(){};
+
+        void setProperty( const std::string& propName, const std::string& propValue){
```

Comment:
Not a bad idea, but we might have to stay with this for the moment. The implementation currently does not own the string that's passed in, leaving it to the caller to free the memory when appropriate. To convert always to lowercase during the set method, will require me to change the behavior, make a copy and then remember to free the memory in case it is overwritten. If we're doing that, then we might as well enhance the implementation to get/set type specific options (setBoolean, setInt, etc) and that would become so much work for not much gain.